simplicity-sys/build.rs

```diff
@@ -0,0 +1,44 @@
+extern crate cc;
+
+use std::path::Path;
+
+fn main() {
+    let simplicity_path = Path::new("depend/simplicity");
+    let files: Vec<_> = vec![
+        "bitstream.c",
+        "dag.c",
+        "deserialize.c",
+        "eval.c",
+        "frame.c",
+        "jets.c",
+        "jets-secp256k1.c",
+        "rsort.c",
+        "sha256.c",
+        "type.c",
+        "typeInference.c",
+        "primitive/elements/env.c",
+        "primitive/elements/exec.c",
+        "primitive/elements/jets.c",
+        "primitive/elements/primitive.c",
+    ]
+    .into_iter()
+    .map(|x| simplicity_path.join(x))
+    .collect();
+    let test_files: Vec<_> = vec![
```

Comment:
In cdee8cba155909bf562aa287b1a0ef3d5318e471:

I don't think we need this array (or these files, for that matter).

Reply:
Ah -- you have unit tests in -sys that use them. In this case I think you could gate this array with `#[cfg(test)]` so that it's not built when simplicity-sys is used as a dependency.

Reply:
We could reduce the built files to a minimum. For instance, in order to construct C-DAGs from binary, we definitely need `decodeMallocDag()` from `dag.c`.

Reply:
The point of this PR is to expose (1) a way to decode DAGs from bytes in C, and (2) test programs with hardcoded bytes and Merkle roots. This is why we need `test_files`. My original plan was to test Rust against C, but this has to be done in a future PR due to outdated jets.